### PR TITLE
test(#101): adopt Faker builders for test data across all test types

### PR DESCRIPTION
**Pull request by RudoiDmytro**
## What & why

Adopts [`@faker-js/faker`](https://fakerjs.dev/) as the standard way to generate test data across all test types, replacing hardcoded magic-value fixtures (emails, names, passwords, ids, tokens) with generated, intention-revealing data. Implements issue #101.

## How

- **Shared builders** in `tests/builders/` (`buildUser`, `buildCredentials`, `buildEmail`, `buildFullName`, `buildPassword`, `buildToken`, `buildUserId`, `buildLoginResponse`, `buildRegistrationResponse`, `buildCreateUserInput`, `buildGraphqlUser`), imported via a new `@tests/*` path alias. Every builder returns **domain-valid data by construction** (passes the app's email / password / name validators) and accepts an `overrides` object.
- **Deterministic seeding** — `seedFaker()` wired into every runner's setup (`jest.setup.ts`, `tests/integration/setup.ts`, new `tests/apollo-server/setup.ts`, and the Playwright e2e/visual shared constants). Default `DEFAULT_FAKER_SEED`, override with `FAKER_SEED=<integer>`; the active seed is reported once per worker so failures are reproducible.
- **Converted fixtures** across unit, integration, Apollo-server, and e2e/visual tests to use the builders. The integration mock server now exposes its generated defaults (`defaultLoginResponse`, `defaultGraphqlUser`, `defaultClientMutationId`) so tests assert against those instead of literals.
- **Faker ESM** transpiled for Jest (faker v10 is ESM-only) via a scoped `babel-jest` + `@babel/preset-env` transform and a `transformIgnorePatterns` exception.
- **Docs** — full convention + review guideline in `agents.md`; pattern reference in `CLAUDE.md`.

## Scope decisions

- Intentional edge-case / invalid literals (malformed emails, boundary passwords), golden text, config values, URLs, error codes, i18n strings, and mock sentinels are **kept** — they are the test case or a fixed contract, not arbitrary fixtures.
- Visual snapshots are unaffected: the success/error notification overlays the form (only i18n strings are in-frame), and deterministic seeding keeps generated data stable.
- The "discourage new hardcoded fixtures" item is a **documented review guideline**, not an ESLint rule — "arbitrary fixture vs intentional literal" is a semantic distinction a static rule can't reliably make (same reasoning as the #89 class-only gate).

## Verification

- `make test-unit-client`, `make test-unit-server`, `make test-integration` — green.
- `make format`, `make lint` — clean.

Closes #101

---
## Summary by cubic
Standardizes test data generation with `@faker-js/faker` builders and deterministic seeding across unit, integration, Apollo-server, and e2e tests to make tests clearer and reproducible. Adds an e2e mock pass-through for `clientMutationId` and tightens threading assertions. Implements issue #101.

- **Refactors**
  - Added shared Faker-backed builders in `tests/builders` (user/auth/GraphQL); domain-valid by construction, `overrides` supported; import via `@tests/*`.
  - Deterministic seeding in all runners (`jest.setup.ts`, `tests/integration/setup.ts`, `tests/apollo-server/setup.ts`, Playwright); default `DEFAULT_FAKER_SEED`, override with `FAKER_SEED`; seed logged once per worker. Enabled ESM-only Faker in Jest via `babel-jest` + `@babel/preset-env` and a `transformIgnorePatterns` exception.
  - Converted tests to builders; integration mock server now exports `defaultLoginResponse`, `defaultGraphqlUser`, `defaultClientMutationId`.
  - E2E success mock now echoes the request `clientMutationId` to match the real API.
  - Strengthened `AuthRepositoryImpl` happy-path tests to assert credentials flow to `loginAPI`/`registrationAPI` and the response mappers.
  - Kept intentional literals (invalid inputs, golden text, config, URLs, error codes, i18n, sentinels); visual snapshots unchanged; docs added in `agents.md` and `CLAUDE.md`.

- **Migration**
  - Use `@tests/builders` in new tests; bind a generated value to a const and reuse it in assertions.
  - To reproduce failures, run with `FAKER_SEED=<integer>` (defaults to `DEFAULT_FAKER_SEED`).

<sup>Written for commit 94f499227203c0abe49e7e46c3c94b7d9d059d21. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/VilnaCRM-Org/crm/pull/131?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ let setupFiles: string[] = [];
 if (isIntegration) {
   setupFiles = ['<rootDir>/tests/integration/setup.ts'];
 } else if (TEST_ENV === 'server') {
-  setupFiles = [];
+  setupFiles = ['<rootDir>/tests/apollo-server/setup.ts'];
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply jest.config.ts:37]: Not applied — this misreads the branch. `tests/integration/setup.ts` is wired in the `isIntegration` branch (line 35); line 37 is the `TEST_ENV === 'server'` branch, which correctly loads `tests/apollo-server/setup.ts`. Swapping it would make the server (node) project load the integration setup (msw + node-fetch) and break it. The integration setup does run — the 370 integration tests seed Faker through it.

File: tests/e2e/modules/user/features/auth/components/form-section/auth-forms/utils/responses.ts [hunk 2]
```diff
@@ -9,13 +11,12 @@ export async function successResponse(route: Route): Promise<void> {
...
-          user: {
+          user: buildGraphqlUser({
             email: userData.email,
             initials: userData.fullName,
-            id: '12345',
             confirmed: true,
-          },
-          clientMutationId: '186',
+          }),
+          clientMutationId: buildClientMutationId(),
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/e2e/modules/user/features/auth/components/form-section/auth-forms/utils/responses.ts]: Fixed in 94f49922. `successResponse` now reads the request's `clientMutationId` (`postDataJSON().variables.input.clientMutationId`, which the client sends via `uuidv4()`) and echoes it, falling back to a generated id only when absent.

[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/e2e/modules/user/features/auth/components/form-section/auth-forms/utils/responses.ts]: Fixed in 94f49922 — the mock now echoes the request's `clientMutationId` instead of generating a fresh one.

File: tests/unit/modules/user/features/auth/repositories/auth-repository-impl.test.ts [hunk 3]
```diff
@@ -19,13 +20,15 @@ const makeDeps = (): AuthRepositoryDeps =>
...
+    const token = buildToken();
+    (deps.loginAPI.login as jest.Mock).mockResolvedValue({ token });
+    (deps.loginResponseMapper.map as jest.Mock).mockReturnValue(ok({ token, email }));
     const repo = new AuthRepositoryImpl(deps);
 
-    const result = await repo.login({ email: 'A@B.C', password: 'p' });
+    const result = await repo.login({ email: buildEmail(), password: buildPassword() });
 
-    expect(result).toEqual({ ok: true, value: { email: 'a@b.c', token: 't' } });
+    expect(result).toEqual({ ok: true, value: { email, token } });
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/unit/modules/user/features/auth/repositories/auth-repository-impl.test.ts]: Fixed in 94f49922. Both happy-path tests now use a single `credentials` const and assert the submitted data is threaded: login asserts `loginAPI.login(credentials, …)` and `loginResponseMapper.map(apiResponse, credentials.email)`; register asserts `registrationAPI.register(credentials, …)`. The result reuses `credentials.email`, so the tests now fail if threading regresses.

File: tests/unit/modules/user/features/auth/repositories/auth-repository-impl.test.ts [hunk 4]
```diff
@@ -19,13 +20,15 @@ const makeDeps = (): AuthRepositoryDeps =>
...
-    (deps.loginAPI.login as jest.Mock).mockResolvedValue({ token: 't' });
-    (deps.loginResponseMapper.map as jest.Mock).mockReturnValue(ok({ token: 't', email: 'a@b.c' }));
+    const email = buildEmail();
+    const token = buildToken();
+    (deps.loginAPI.login as jest.Mock).mockResolvedValue({ token });
+    (deps.loginResponseMapper.map as jest.Mock).mockReturnValue(ok({ token, email }));
     const repo = new AuthRepositoryImpl(deps);
 
-    const result = await repo.login({ email: 'A@B.C', password: 'p' });
+    const result = await repo.login({ email: buildEmail(), password: buildPassword() });
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tests/unit/modules/user/features/auth/repositories/auth-repository-impl.test.ts]: Fixed in 94f49922 — the login test reuses one `credentials` const for the input and the mapped result, and additionally asserts `loginResponseMapper.map` receives the submitted `credentials.email` (the real threading point: the repo passes `credentials.email` into the mapper, not the result directly).

File: tsconfig.paths.json [hunk 5]
```diff
@@ -4,7 +4,8 @@
     "paths": {
       "@/*": ["src/*"],
       "@": ["src"],
-      "@auth/*": ["src/modules/user/features/auth/*"]
+      "@auth/*": ["src/modules/user/features/auth/*"],
+      "@tests/*": ["tests/*"]
```
[the remark this answers is not in the document]
Comment (RudoiDmytro) [reply tsconfig.paths.json:8]: Not applied — `@auth/*` (with the `@`) is the existing repo standard, documented in CLAUDE.md (`@auth/* → src/modules/user/features/auth/*`) and used in `jest.config.ts`, `rsbuild.config.ts`, and throughout `src`/`tests`. This PR only added `@tests/*`; it did not change `@auth/*`. Switching to `auth/*` would break every existing `@auth/...` import.

**Review by Kravalg — APPROVED** [review]
